x-pack/plugins/profiling/server/routes/setup.ts

```diff
@@ -86,11 +89,31 @@ export function registerSetupRoute({
           });
         }
 
-        state.data.available = await hasProfilingData({
-          ...setupOptions,
-          client: clientWithProfilingAuth,
-        });
-        if (state.data.available) {
+        const viewerUsersVerifyFunctions = [
+          validateCollectorPackagePolicy,
+          validateSymbolizerPackagePolicy,
+          validateProfilingInApmPackagePolicy,
+        ];
+
+        const viewerUsersPartialStates = await Promise.all([
+          ...viewerUsersVerifyFunctions.map((fn) => fn(setupOptions)),
+          hasProfilingData({
+            ...setupOptions,
+            client: clientWithProfilingAuth,
+          }),
+        ]);
+
+        const viewerUsersMergedState = mergePartialSetupStates(state, viewerUsersPartialStates);
+
+        /*
+         * We need to split the verification steps
+         * because of users with viewer privileges
+         * cannot get the cluster settings
+         */
+        if (
+          areViewerUsersResourcesSetup(viewerUsersMergedState) &&
+          viewerUsersMergedState.data.available
+        ) {
```

Comment:
The value for `has_data` in the response is logically correct, but it had me second guess the value for a second. Could we set `set_data` either to `true` or `mergedStateForViewer.data.available`? I think it would be less likely to cause an issue during a refactor, especially if we moved the logic around.